tests/integration/targets/lambda_layer/tasks/main.yml

```diff
@@ -0,0 +1,222 @@
+      assert:
+        that:
+          - '"layers_versions" in layers'
+          - layers.layers_versions | length == 2
+          - last_version.layer_version.layer_version_arn in layers_version_arns
+          - first_version.layer_version.layer_version_arn in layers_version_arns
+      vars:
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Delete latest layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ last_version.layer_version.version }}"
+        state: absent
+      check_mode: true
+      register: delete_check_mode
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer still contains the same number of versions
+      assert:
+        that:
+          - delete_check_mode is changed
+          - layers.layers_versions | length == 2
+          - last_version.layer_version.layer_version_arn in layers_version_arns
+          - first_version.layer_version.layer_version_arn in layers_version_arns
+      vars:
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Delete latest layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ last_version.layer_version.version }}"
+        state: absent
+      register: delete_layer
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer still contains the same number of versions
+      assert:
+        that:
+          - delete_layer is changed
+          - layers.layers_versions | length == 1
+          - last_version.layer_version.layer_version_arn not in layers_version_arns
+          - first_version.layer_version.layer_version_arn in layers_version_arns
+      vars:
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Delete again the latest layer version (idempotency)
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ last_version.layer_version.version }}"
+        state: absent
+      register: delete_idempotent
+
+    - name: Ensure nothing changed
+      assert:
+        that:
+          - delete_idempotent is not changed
+
+    - name: Delete first layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ first_version.layer_version.version }}"
+        state: absent
+      register: delete_layer
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer does not exist anymore
+      assert:
+        that:
+          - delete_layer is changed
+          - layers.layers_versions | length == 0
+
+  always:
+    - name: Delete temporary directory
```

Comment:
Shouldn't we also delete the lambda layers?

Reply:
we can even if it is supposed to be deleted by the test case, but I will add an extra check